es/es-server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -1390,6 +1390,10 @@ public void close() {
             release();
         }
 
+        public boolean exists() {
```

Comment:
@seut do you mind if we keep the `exists` method or better to create some `assertXYZ` in the test class to check whether it exists?

Reply:
Isn't `exists` defined by the tests (or GetResult.NO_EXISTS) to be always false? wouldn't this then assert that a false is a false?

Reply:
I had removed `exists` because there was an implicit contract/relation between `exists` and `docIdAndVersion`, which cannot be asserted by the compiler, so to be safe you'd always have to to assert that if `exists` is true, `docIdAndVersion` must not be null, which is tedious. 

Please don't re-add it

Reply:
Ok I've wrongly assumed that `NO_EXISTS` is used by test, but thats not true, the engine will use/return it. So yes let's keep this method..

Reply:
Sorry didn't see @mfussenegger comment, I agree with him, let's not re-add it.

Reply:
ok, so i see two options:
1. use `assertThat` with `equalsTo`, but then we would have implement equals and hashCode in some classes.
2. create an `assertGetResultExist` in the test class that would check whether fields are not nulls.

would the second option be fine?

Reply:
+1 for `assertGetResultExist`.

Reply:
Why not `assertThat(...docIdAndVersion(), Matchers.notNullValue())` ?

Reply:
yep 👍 searcher is not needed.